test/jdk/java/util/stream/BuiltInGatherersTest.java

```diff
@@ -245,34 +253,43 @@ public void testMapConcurrentAPIandContract(Config config) throws InterruptedExc
         }
 
         // Test cancellation after exception during processing
-        if (config.streamSize > 2) { // We need streams of a minimum size to test this
-            final var firstLatch = new CountDownLatch(1);
-            final var secondLatch = new CountDownLatch(1);
-            final var cancellationLatch = new CountDownLatch(config.streamSize - 2); // all but two will get cancelled
+        // Only use reasonably sized streams to avoid excessive thread creation
+        if (config.streamSize > 2 && config.streamSize < 100) {
```

Comment:
@AlanBateman Decided to not run the cancellation tests for the larger streams as it creates a bunch of extra resource usage which makes GHA really unhappy.

Reply:
Okay but it makes me wonder if this test should have its use method source with the stream sizes that are sensible to test.